src/api/voting.ts

```diff
@@ -206,41 +229,79 @@ export class VotingApi {
                 result.processId = "0x" + result.processId
                 return result
             })
-            .catch((error) => {
-                const message = error.message ? "Could not retrieve the process info: " + error.message : "Could not retrieve the process info"
-                throw new Error(message)
-            })
+    }
+
+    /**
+     * Fetch the full state of the given processId on the process archive
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    private static getArchiveProcessState(processId: string, gateway: IGateway | IGatewayPool, errorMessage: string): Promise<DVoteGatewayResponseBody> {
+        return GatewayArchive.getProcessFromArchive(processId, gateway, errorMessage)
+            .then((response: DVoteGatewayResponseBody) => response)
     }
 
     /**
      * Fetch the Vochain headers of the given processId on the Vochain. This operation is more lightweight than getProcessInfo
      * @param processId
      * @param gateway
+     * @param checkArchive
      */
-    static getProcessSummary(processId: string, gateway: IGateway | IGatewayPool): Promise<IProcessSummary> {
+    static getProcessSummary(processId: string, gateway: IGateway | IGatewayPool, checkArchive: boolean = false): Promise<IProcessSummary> {
         if (!processId) return Promise.reject(new Error("Empty process ID"))
         else if (!gateway) return Promise.reject(new Error("Invalid Gateway object"))
 
         return gateway.sendRequest({ method: "getProcessSummary", processId })
+            .catch((error) => {
+                const message = error.message ? "Could not retrieve the process info: " + error.message : "Could not retrieve the process info"
+                if (checkArchive && message.includes("No data found for this key")) {
+                    return this.getArchiveProcessSummary(processId, gateway, message)
+                }
+                throw new Error(message)
+            })
             .then((response) => {
-                if (!response.ok) throw new Error(response.message || null)
                 const { processSummary } = response
 
                 return {
                     entityId: "0x" + processSummary.entityId,
                     envelopeHeight: processSummary.envelopeHeight,
-                    status: VochainProcessStatus[processSummary.state as string],
+                    status: processSummary.status ? processSummary.status : VochainProcessStatus[processSummary.state as string],
```

Comment:
Should we check for this to be explicitly a number? If status is `0` (READY), then we will miss it.

```suggestion
                    status: typeof processSummary.status === "number" ? processSummary.status : VochainProcessStatus[processSummary.state as string],
```

Reply:
Nice catch